umpost/conversion_driver_esm1p5.py

```diff
@@ -268,6 +268,45 @@ def format_failures(failed, quiet):
             yield failure_report
 
 
+def _resolve_path(path):
+    """
+    Resolve path for use in comparison. Ensure that symlinks, relative paths,
+    and home directories are expanded.
+    """
+    return os.path.realpath(os.path.expanduser(path))
+
```

Comment:
I'm wondering if this is overkill compared to just using `pathlib`'s `Path.resolve()` method. My main concern is making sure different representations of the same path are properly handled. `Path.resolve()` doesn't treat e.g. `~/a/b` and `/<home-directory>/a/b` the same, hence the use of `os.path.expanduser()`.

This is unlikely to come up in practice however, as the output paths are created elsewhere in the conversion driver, and I don't think there would be any way for one output to be given `~/a/...` and another to be given `/<home-directory>/a/...`.

Reply:
I don't mind being thorough if it's not an expensive operation (not being done thousands of times).